packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1111,11 +1110,13 @@ new PrismaClient({
 
         if (globalThis.NOT_PRISMA_DATA_PROXY) {
           // async scope https://github.com/prisma/prisma/issues/3148
-          const resource = new AsyncResource('prisma-client-request')
-          return resource.runInAsyncScope(() => consumer(params))
+          return await new AsyncResource('prisma-client-request').runInAsyncScope(() => {
```

Comment:
Why is await needed here now?

Reply:
Some logic changed, but basically we want async operations to be `try/catch` before the return.